### PR TITLE
Proposal: add Meson build support

### PR DESCRIPTION
**Pull request by kirbyfan64**
## Synopsis

This is the thing that happens when you're trying to change something and get tired of dealing with weird stuff/rpath crud, autotools randomly reconfiguring, and slow-ish builds, so you decide to spend an inordinate amount of time sidetracked on build systems.

AFAIK this supports *everything* the autotools build system supports, except it's roughly half the size, an order of magnitude faster, and a lot cleaner.

## Why Meson?

Meson is quickly getting popular in the world of Unix-y build systems. It's now used by:

- [systemd](https://lists.freedesktop.org/archives/systemd-devel/2017-April/038582.html)
- [Xserver](https://lists.x.org/archives/xorg-devel/2017-April/053509.html)
- [GTK+](https://mail.gnome.org/archives/gtk-devel-list/2017-August/msg00028.html)
- [libepoxy](https://www.bassi.io/articles/2017/02/11/epoxy/)
- [libdrm](https://lwn.net/Articles/718351/)
- [Wayland](https://lists.freedesktop.org/archives/wayland-devel/2018-April/037817.html)

Basically, take the autotools configuration support, but throw away the weird macro language and makefiles, replacing them with a Python-ish-inspired DSL...and you've got Meson!

### CODE REVIEW

**Comment by Aetf** [comment]
Thanks for the contribution! I really appreciate it.

It's true that fighting with autotools is not so fun. But I think it's better to use something I'm fluent in, and I've decided to switch to cmake soon. So sorry, I'm closing this.
